levelmap.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Siôn le Roux.  All rights reserved.
+// Use of this source code is subject to an MIT-style
+// licence which can be found in the LICENSE file.
+
+package main
+
+import (
+	"image"
+	"math"
+
+	"github.com/fzipp/astar"
+)
+
+type LevelMap [][]int
+
+
+func CreateMap(w, h int) LevelMap {
+	lmap := make([][]int, h)
+	for i := range lmap {
+		lmap[i] = make([]int, w)
+	}
+	
+	return lmap
+}
+
+// Neighbours implements the astar.Graph interface
+func (m LevelMap) Neighbours(p image.Point) []image.Point {
+	offsets := []image.Point{
+		image.Pt(0, -1), // North
+		image.Pt(1, 0),  // East
+		image.Pt(0, 1),  // South
+		image.Pt(-1, 0), // West
+	}
+	offsetsDiag := []image.Point{
+		image.Pt(1, -1),  // NorthEast
+		image.Pt(1, 1),   // SouthEast
+		image.Pt(-1, 1),  // SouthWest
+		image.Pt(-1, -1), // NorthWest
+	}
+	res := make([]image.Point, 0, 8)
+	
+	// Check avaialable diagonal neighbours (free only if there is a wide enough corridor)
+	for _, off := range offsetsDiag {
```

Comment:
This is just a nitpick, but naming this variable `off` completely threw me off because I thought it was going to represent some boolean state, like is it on/off, but it's actually short for offset. Maybe `offset` or `o` would avoid this ambiguity?

Reply:
OK